build_tools/torchscript_e2e_heavydep_tests/facebook_dlrm.py

```diff
@@ -0,0 +1,839 @@
+import torch
+import torch_mlir
+import numpy as np
+import torch.nn as nn
+from typing import Dict, List
+from numpy import random as ra
+from torch.utils.data import Dataset
+
+from torch_mlir_e2e_test.torchscript.framework import TestUtils
+from torch_mlir_e2e_test.torchscript.registry import register_test_case
+from torch_mlir_e2e_test.torchscript.annotations import annotate_args, export
+
+class DLRM_Net(nn.Module):
```

Comment:
can we please use an existing library rather than open-coding this ourselves? We want this to be representative of what users would see.

Reply:
IIRC, this is more or less lifted from the existing reference implementation, no? https://github.com/facebookresearch/dlrm/blob/main/dlrm_s_pytorch.py

As an aside: that probably means it should really be licensed accordingly (i.e., (c) facebook, MIT license), since it's derived code.

Reply:
I'm not specifically familiar, but yes, the code does look extremely similar.

We should use that library since that is what users will presumably be using. We can add that library as a dependency here: https://github.com/llvm/torch-mlir/blob/5bd9362c61828870cf3c4097c718ea34f475b11a/build_tools/torchscript_e2e_heavydep_tests/generate_serialized_tests.sh#L26

Reply:
yes, its taken from the facebook dlrm repo: https://github.com/facebookresearch/dlrm
Apologies, will add the correct license.

Reply:
Can you please depend on that DLRM library instead of copying their code?

Reply:
which library ?

Reply:
can you pip install https://github.com/facebookresearch/dlrm ?

Reply:
not that I am aware of. doesn't say on their README.

Reply:
I think @silvasean means can we `pip install facebookdlrm` or equivalent.  I don't think they have a setup.py even just a bunch of .py scripts. I tried to `pip install -e git+https://...` but it wasn't happy without a setup.py

Reply:
I see, also @silvasean @powderluv  I had to make many changes in their source code to be able to get TorchScript via torch.jit.trace. So I don't know how that would work even if it was pip install-able ...

Reply:
I think their code in https://github.com/facebookresearch/dlrm  is meant to be the reference DLRM for the industry since each vendor had different variants (https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/Recommendation/DLRM and https://github.com/jithunnair-amd/dlrm). Unfortunately the reference DLRM code doesn't lower cleanly without some changes to the model and the authors suggest we make such changes https://github.com/facebookresearch/dlrm/issues/132#issuecomment-708902880

I have been pushing the team to upstream model variants with fixes (just like HuggingFace carries lowering model fixes) into torch-mlir e2e tests but if we want to only use "stock" models that is ok too. We can carry the fixed up / lowerable models downstream in SHARK model/tests suites and report reproductions of issues upstream.

Reply:
My main concern is that I want this to be representative of how a user would actually want to use DLRM via Torch-MLIR -- would they really copy-paste that reference code? It's not really Torch-MLIR's job to vendor a DLRM implementation, let alone having such a vendored implementation only as part of our test-suite and not even exposed to users (requiring yet another copy-paste telephone game).

Reply:
closing this issue and move dlrm to SHARK model/tests suites